### PR TITLE
Events

### PR DESCRIPTION
**Pull request by tta2yta**
In this project, the calculator logical operations are implemented in the math magician application

Please review my code before I merge to develop the branch.

Thank you.

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,51 +1,118 @@
...
+  isNumber=(item) => !!item.match(/[0-9]+/)
+
   render() {
+    const { next } = this.state;
     return (
       <div className="calculator">
         <div className="calculator-body">
           <div className="input-val">
-            <input type="text" value="0" />
+            <input type="text" value={next} onChange={this.handleChange} />
```
Comment (Davidosky007) [inline src/components/calculator.js:83]: * I get errors in the console when i click on operation buttons (ie +, -) 
image

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      }
+      return { total: `${obj.total}.` };
+    }
+    return { total: '0.' };
+  }
+
+  if (buttonName === '=') {
+    if (obj.next && obj.operation) {
+      return {
+        total: operate(obj.total, obj.next, obj.operation),
```
Comment (Davidosky007) [inline src/logic/calculate.js:71]: * When i perform operations divided by zero (ie 7 / 0) i get this error kindly fix 🙏 
image

File: src/components/calculator.js [hunk 3]
```diff
@@ -1,51 +1,118 @@
...
+      console.log(res);
+      console.log(this.state);
+    }
+
+    if (e.target.outerText === '=') {
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ ...res });
+      this.setState({ next: res.total });
+      console.log(res);
```
Comment (Davidosky007) [inline src/components/calculator.js]: * I suggest you remove `console.log` from your codebase as it is not professional and might log sensitive information to  console therefore exposing it to the user [Read more](https://medium.com/@hfally/deactivate-console-log-on-production-why-and-how-d0345abbe71)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @tta2yta 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
* Kindly remove `console.log` statements from your codebase it makes your code bulky 🙏 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashbaord. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**